docs/hazmat/primitives/mac/poly1305.rst

```diff
@@ -85,3 +85,44 @@ messages allows an attacker to forge tags. Poly1305 is described in
 
         :return bytes: The message authentication code as bytes.
         :raises cryptography.exceptions.AlreadyFinalized:
+
+    .. classmethod:: generate_tag(key, data)
+
+        A single step alternative to do sign operations. Returns the message
+        authentication code as ``bytes`` for the given ``key`` and ``data``.
+
+        :param key: Secret key as ``bytes``.
+        :type key: :term:`bytes-like`
+        :param data: The bytes to hash and authenticate.
+        :type data: :term:`bytes-like`
+        :return bytes: The message authentication code as bytes.
+        :raises cryptography.exceptions.UnsupportedAlgorithm: This is raised if
+            the version of OpenSSL ``cryptography`` is compiled against does not
+            support this algorithm.
+        :raises TypeError: This exception is raised if ``key`` or ``data`` are
+            not ``bytes``.
+
+        >>> poly1305.Poly1305.generate_tag(key, b"message to authenticate")
+        b'T\xae\xff3\xbdW\xef\xd5r\x01\xe2n=\xb7\xd2h'
+
+    .. classmethod:: verify_tag(key, data, tag)
+
+        A single step alternative to do verify operations. Securely compares the
+        MAC to ``tag``, using the given ``key`` and ``data``.
+
+        :param key: Secret key as ``bytes``.
+        :type key: :term:`bytes-like`
+        :param data: The bytes to hash and authenticate.
+        :type data: :term:`bytes-like`
+        :param bytes tag: The bytes to compare against.
+        :raises cryptography.exceptions.UnsupportedAlgorithm: This is raised if
+            the version of OpenSSL ``cryptography`` is compiled against does not
+            support this algorithm.
+        :raises TypeError: This exception is raised if ``key``, ``data`` or
+            ``tag`` are not ``bytes``.
+        :raises cryptography.exceptions.InvalidSignature: If tag does not match.
+
+        >>> poly1305.Poly1305.verify_tag(key, b"message to authenticate", b"an incorrect tag")
```

Comment:
I'm not sure this is particularly useful as documentation for the exception. We typically don't put these in our docs (and if we do want to do it then we want it to be doctest to avoid bitrot in examples)

Reply:
I based this off the `update` `verify` docs where there's an example of the exception being thrown https://cryptography.io/en/latest/hazmat/primitives/mac/poly1305/#cryptography.hazmat.primitives.poly1305.Poly1305.

I've changed this to a doctest for now. Should we go ahead and remove the example?

Reply:
If we've put it elsewhere and it works in doctest then I'm fine with it (and it appears it does work in doctest)